File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/model/coupon/CouponCourseData.java
package com.huythanh0x.udemycoupons.model.coupon;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.time.LocalDateTime;

@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class CouponCourseData {
    @Id
    private int courseId;
    private String category;
    private String subCategory;
    private String title;
    private int contentLength;
    private String level;
    private String author;
    private float rating;
    private int reviews;
    private int students;
    private String couponCode;
    private String previewImage;
    private String couponUrl;
    @Column(name = "expired_date", nullable = false)
    private Instant expiredDate;
    private int usesRemaining;
    private String heading;
    @Column(columnDefinition = "TEXT")
    private String description;
    private String previewVideo;
    private String language;
    @Builder.Default
    @Column(name = "is_new", nullable = false)
    private boolean isNew = true;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/model/coupon/CouponCourseData.kt
package com.thanh0x.coursedeal.model.coupon

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Suppress("LongParameterList")
class CouponCourseData(
    @Id
    var courseId: Int = 0,
    var category: String? = null,
    var subCategory: String? = null,
    var title: String? = null,
    var contentLength: Int = 0,
    var level: CourseLevel? = null,
    var author: String? = null,
    var rating: Float = 0f,
    var reviews: Int = 0,
    var students: Int = 0,
    var couponCode: String? = null,
    var previewImage: String? = null,
    var couponUrl: String? = null,
    @Column(name = "expired_date")
    var expiredDate: Instant? = null,
    var usesRemaining: Int = 0,
    var heading: String? = null,
    @Column(columnDefinition = "TEXT")
    var description: String? = null,
    var previewVideo: String? = null,
    var language: String? = null,
    @Column(name = "is_new", nullable = false)
    var isNew: Boolean = true,
    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    var createdAt: LocalDateTime? = null,
    @UpdateTimestamp
    @Column(name = "updated_at")
    var updatedAt: LocalDateTime? = null,
)
